pkg/agent/cniserver/interface_configuration_linux.go

```diff
@@ -0,0 +1,342 @@
+// Copyright 2020 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build linux
+
+package cniserver
+
+import (
+	"fmt"
+	"net"
+	"time"
+
+	cnitypes "github.com/containernetworking/cni/pkg/types"
+	"github.com/containernetworking/cni/pkg/types/current"
+	"github.com/containernetworking/plugins/pkg/ip"
+	"github.com/containernetworking/plugins/pkg/ipam"
+	"github.com/containernetworking/plugins/pkg/ns"
+	"github.com/j-keck/arping"
+	"github.com/vishvananda/netlink"
+	"k8s.io/klog"
+
+	"github.com/vmware-tanzu/antrea/pkg/agent/util"
+	"github.com/vmware-tanzu/antrea/pkg/agent/util/ethtool"
+	cnipb "github.com/vmware-tanzu/antrea/pkg/apis/cni/v1beta1"
+	"github.com/vmware-tanzu/antrea/pkg/ovs/ovsconfig"
+)
+
+type ifConfigurator struct {
+	ovsDatapathType string
+}
+
+func newInterfaceConfigurator(ovsDatapathType string) interfaceConfigurator {
```

Comment:
Normally the newXX function should return the concrete types (the pointer to the struct or the struct itself), not the interface:
1. you may use this returned value to test other functions not in the interface.
2. a struct may implement more than one interface, its consumers can use the desired interface to receive the pointer.

See also https://github.com/golang/go/wiki/CodeReviewComments#interfaces